e3sm_diags/parameter/core_parameter.py

```diff
@@ -1,18 +1,59 @@
-from __future__ import print_function
+import copy
+from typing import Dict, List, Optional
 
-import cdp.cdp_parameter
 
-
-class CoreParameter(cdp.cdp_parameter.CDPParameter):
```

Comment:
Most of the "important" changes are in the `core_parameter.CoreParameter` class.

Changes include:
 - Removing `CDPParameter` sub-classing
 - Organize attributes with comments
   - Adding missing attributes that are set on `CoreParameter` objects in various locations of the codebase
   - Add type annotations for attributes and variables
 - Add `__add__` which is ported from `CDPParameter`
- Update `check_values()` and its `must_have_params` list